src/controllers/webhook.controller.ts

```
@@ -64,20 +318,36 @@ export class WebhookController {
 
-      // Verify signature
-      const payload = req.body as WebhookPayload;
-      const isValid =
-        await this.webhookNotificationService.sendWebhookNotification(
-          webhook.url,
-          payload,
-          merchantId,
+      // Verify signature if we have a secret key
+      if (webhook.secretKey && signature) {
+        // Use rawBody if available (needs express configuration), otherwise fallback to body
+        const rawBody = (req as any).rawBody;
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)